dev/MRTCore/mrt/Microsoft.Windows.ApplicationModel.Resources/src/ApplicationLanguages.cpp

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation and Contributors.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "ApplicationLanguages.h"
+#include "ApplicationLanguages.g.cpp"
+
+namespace winrt::Microsoft::Windows::ApplicationModel::Resources::implementation
+{
+    hstring ApplicationLanguages::m_language;
+
+    void ApplicationLanguages::PrimaryLanguageOverride(hstring language)
```

Comment:
Is this file from the generated skeleton from C++/WinRT or hand crafted? Generated files usually have string parameters as 'hstring const& blah' (`hstring` will cause a deep copy on each call)

After compiling your .idl grab the code from `d:\source\repos\windowsappsdk\obj\Debug\x64\WindowsAppRuntime_DLL\Generated Files\sources\Microsoft.Windows.ApplicationModel.Resources.ApplicationLanguages.cpp` and matching .h, and follow their contained directions.

Reply:
my bad. Fixed. Thank you